src/deluge/gui/ui/keyboard/layout/norns.h

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright © 2016-2023 Synthstrom Audible Limited
+ *
+ * This file is part of The Synthstrom Audible Deluge Firmware.
+ *
+ * The Synthstrom Audible Deluge Firmware is free software: you can redistribute it and/or modify it under the
+ * terms of the GNU General Public License as published by the Free Software Foundation,
+ * either version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with this program.
+ * If not, see <https://www.gnu.org/licenses/>.
+*/
+
+#pragma once
+
+#include "gui/ui/keyboard/layout.h"
+
+namespace deluge::gui::ui::keyboard::layout {
+
+constexpr int32_t kMinNornsRowInterval = 1;
+constexpr int32_t kMaxNornsRowInterval = 16;
+
+class KeyboardLayoutNorns : public KeyboardLayout {
+public:
+	KeyboardLayoutNorns() {}
+	virtual ~KeyboardLayoutNorns() {}
+
+	virtual void evaluatePads(PressedPad presses[kMaxNumKeyboardPadPresses]);
```

Comment:
If these are overriding a parent's virtual method they should be marked override instead of virtual.

Reply:
Fixed in all layouts :)